servers/src/main/java/tachyon/master/lineage/LineageMaster.java

```diff
@@ -343,8 +343,10 @@ private void asyncCompleteFileFromEntry(AsyncCompleteFileEntry entry) {
    */
   public synchronized LineageCommand lineageWorkerHeartbeat(long workerId,
       List<Long> persistedFiles) throws FileDoesNotExistException, InvalidPathException {
-    // notify checkpoint manager the persisted files
-    persistFiles(workerId, persistedFiles);
+    if (!persistedFiles.isEmpty()) {
+      // notify checkpoint manager the persisted files
+      persistFiles(workerId, persistedFiles);
```

Comment:
I noticed persistFiles() is public, but only this method calls it. Should that be private?

I ask because if somehow something else calls persistFiles(), it could still generate empty journal entries, since the check is done at this level.